### ISSUE TITLE
Sort field not set properly

### ISSUE THREAD

**Issue by johannes-richter** [directus/directus#20513]
### Describe the Bug

When creating a new collection with the sort field option, Directus does not create the fields properly.
I have the sort field set and the request does not give me an error:

```
{
  "collection": "Translated_status_messages",
  "fields": [
    {
      "field": "id",
      "type": "uuid",
      "meta": {
        "hidden": true,
        "readonly": true,
        "interface": "input",
        "special": [
          "uuid"
        ]
      },
      "schema": {
        "is_primary_key": true,
        "length": 36,
        "has_auto_increment": false
      }
    },
    {
      "field": "status",
      "type": "string",
      "meta": {
        "width": "full",
        "options": {
          "choices": [
            {
              "text": "$t:published",
              "value": "published"
            },
            {
              "text": "$t:draft",
              "value": "draft"
            },
            {
              "text": "$t:archived",
              "value": "archived"
            }
          ]
        },
        "interface": "select-dropdown",
        "display": "labels",
        "display_options": {
          "showAsDot": true,
          "choices": [
            {
              "text": "$t:published",
              "value": "published",
              "foreground": "#FFFFFF",
              "background": "var(--theme--primary)"
            },
            {
              "text": "$t:draft",
              "value": "draft",
              "foreground": "#18222F",
              "background": "#D3DAE4"
            },
            {
              "text": "$t:archived",
              "value": "archived",
              "foreground": "#FFFFFF",
              "background": "var(--theme--warning)"
            }
          ]
        }
      },
      "schema": {
        "default_value": "draft",
        "is_nullable": false
      }
    },
  // ... shortened
    {
      "field": "translations",
      "type": "alias",
      "meta": {
        "interface": "translations",
        "special": [
          "translations"
        ],
        "options": {
          "defaultLanguage": "en-US",
          "languageField": "code",
          "languageDirectionField": "reading_direction"
        }
      }
    }
  ],
  "schema": {},
  "meta": {
    "archive_field": "status",
    "archive_value": "archived",
    "unarchive_value": "draft",
    "singleton": false
  }
}
```

This gives me a 200. However, when I try to insert an item into that collection, I get the following error

```
{
    "errors": [
        {
            "message": "select top (@p0) [Translated_status_messages].[id], [Translated_status_messages].[status], [Translated_status_messages].[author], [Translated_status_messages].[date_created], [Translated_status_messages].[date_published], [Translated_status_messages].[lifespan_from], [Translated_status_messages].[lifespan_to], [Translated_status_messages].[message_type], [Translated_status_messages].[link_internal], [Translated_status_messages].[link_external], [Translated_status_messages].[target_component] from [Translated_status_messages] where [Translated_status_messages].[id] = @p1 order by [Translated_status_messages].[sort] asc - Invalid column name 'sort'.",
            "extensions": {
                "code": "INTERNAL_SERVER_ERROR"
            }
        }
    ]
}
```

The funny thing is: Directus will still write that item into the database! And in the items preview of the collection, the items can be seen after a page reload. However, once I click an item and the GETter is trigged, it comes back with a similar response text and a 500. 

I fixed this by manually adding the "sort" field into the table in the database. But this needs some attention I guess, it's ruining my automation flow of creating a ready-to-use Directus instance after the upgrade to the latest version. ( I made a jump form 10.6 though )
On another try I completely deleted the meta tag from the collections call and for some reason I don't get that error anymore. No changes made to any other request or item creation request. And, no sort column in the database also. Very strange :-)

### To Reproduce

Create a collection that does not have a sort field and include a meta tag inside the request.

### Directus Version

v10.8.1

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by br41nslug** [comment]
Is the sort field in the redacted `// ... shortened` part? i neither see the sort field definition in the fields list nor do i see the `sort_field` meta option being set there 🤔 When manually configuring a collection sort you will need to make sure the field exists.

I've created a collection based on the snippet provided and was not able to reproduce the issue with the given data. Closing this issue for now but am happy to re-open it once enough information for reproduction is available.

**Comment by johannes-richter** [comment]
Hey. Sorry I made a language mistake: I meant to create a collection without the sort option! Sorry for the confusion. In short, even when I wasn't ticking the sort field or not using it inside the /collections call, Directus was looking for the field when saving or retrieving items.
Also, I got it to work with not including any data inside the meta tag of the call. So we can keep it closed I guess. Thanks for having a look into it.
